src/Moq/Protected/ProtectedAsMock.cs

```diff
@@ -305,7 +322,7 @@ private PropertyInfo FindCorrespondingProperty(PropertyInfo duckProperty)
 			{
 				var candidateTargetProperties =
 				    this.targetType
-				    .GetProperties(BindingFlags.NonPublic | BindingFlags.Instance)
+				    .GetProperties(BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance)
```

Comment:
I'm not sure I understand this change.

* How is it related to the addition of `SetupSet`, `VerifySet`?
* My memory may be a bit rusty, but this looks incorrect to me: we're looking for a `protected` member, so why do we need `BindingFlags.Public`?

Reply:
> How is it related to the addition of SetupSet, VerifySet?

This is the fix for virtual property with protected accessor and applies equally for SetupSet as it does for SetupGet.

> My memory may be a bit rusty, but this looks incorrect to me: we're looking for a protected member, so why do we need BindingFlags.Public?

Yes we are looking for a protected accessor of a public property.

Reply:
This BindingFlags change makes ProtectedAsMock behave the same as ProtectedMock

https://github.com/moq/moq4/blob/a6fde8b6d79a7437bf642d115785b97f40779b6a/src/Moq/Protected/ProtectedMock.cs#L346

```
		private static PropertyInfo GetProperty(string propertyName)
		{
			return typeof(T).GetProperty(
				propertyName,
				BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.Public);
		}
```

Reply:
Without reviewing the existing code I couldn't say in which places `ProtectedAsMock` needs to be the same as `ProtectedMock` and where it may legitimately diverge, but your first explanation (part of your bug fix) is good enough for me! :-)